metrics/measured_http/http.go

```diff
@@ -62,11 +62,37 @@ func (h *MeasuredHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	begin := h.clk.Now()
 	rwws := &responseWriterWithStatus{w, 0}
 
+	// Use the method string only if it's a recognized HTTP method. This avoids
+	// ballooning timeseries with invalid methods from public input.
+	var method string
+	switch r.Method {
+	case http.MethodGet:
+		fallthrough
+	case http.MethodHead:
+		fallthrough
+	case http.MethodPost:
+		fallthrough
+	case http.MethodPut:
+		fallthrough
+	case http.MethodPatch:
+		fallthrough
+	case http.MethodDelete:
+		fallthrough
+	case http.MethodConnect:
+		fallthrough
+	case http.MethodOptions:
+		fallthrough
+	case http.MethodTrace:
+		method = r.Method
+	default:
+		method = "unknown"
```

Comment:
Nope, Nginx doesn't 405, Boulder does (I think most things don't bother implementing 405, I've only ever seen it in Boulder). And Akamai appears to let these through too, since we see them in practice.